### ISSUE TITLE
`--output-signature` and `--output-certificate` flags are greedy

### ISSUE THREAD

**Issue by di** [sigstore/sigstore-python#122]
**Description**

When trying to use the `--output-*` flags with default values, invocation fails because they try to consume the filename of the artifact to be signed:

```
$ python -m sigstore sign --output-signature --output-certificate hi.txt
Usage: python -m sigstore sign [OPTIONS] FILE [FILE ...]
Try 'python -m sigstore sign --help' for help.

Error: Missing argument 'FILE [FILE ...]'.
```

**Comment by woodruffw** [comment]
@tetsuo-cpp This is a good one to look into. There _might_ be a way to make these options non-greedy, although a quick look through `click`'s documentation didn't reveal anything to me 😞

**Comment by di** [comment]
I suppose there's no way to distinguish between specifying a signature filename:

```
$ python -m sigstore sign --output-signature <signature_file> <file_to_sign>
```

and trying to use the default / empty flag with multiple files:

```
$ python -m sigstore sign --output-signature <file_to_sign> <another_file_to_sign>
```

Given that we still want to support outputting verification materials when using globbing/directories without needing to specify filenames, we probably need to figure out different UX for "output the verification materials with default filenames".

**Comment by jku** [comment]
I agree options that may or may not take arguments are mostly traps for users... but the standard double-dash does work in click:

`sigstore sign --output-signature --output-certificate -- hi.txt`

**Comment by woodruffw** [comment]
Yeah, I think we should scrap the optional value support here. How does this sound?

* We retain `--output-*` with explicit values,  e.g. `sigstore sign foo --output-signature foo.sig`
* We continue to forbid `--output-*` when multiple input files are passed
* Instead of taking an optional value, we add some kind of `--write` or `--default-outputs` flag that behaves the way "bare" `--output-*` currently behaves

This allows us to retain all current functionality without ambiguity.

**Comment by di** [comment]
Maybe just `--output`?

**Comment by woodruffw** [comment]
Sounds good. I can roll that together with #123.
